armi/reactor/assemblies.py

```diff
@@ -180,6 +180,14 @@ def add(self, obj: blocks.Block):
         .. impl:: Assemblies are made up of type Block.
             :id: I_ARMI_ASSEM_BLOCKS
             :implements: R_ARMI_ASSEM_BLOCKS
+
+            Adds a unique Block to the top of the Assembly. If the Block already
+            exists in the Assembly, an error is raised in
+            :py:meth:`armi.reactor.composites.Composite.add`.
+            The spatialLocator of the Assembly is automatically updated
+            to account for the new Block. The spatial locator for the Block is
+            also updated in ``reestablishBlockOrder``. The axial mesh and other
+            block geometry parameters are updated in ``calculateZCoords``.
```

Comment:
We may want to add the following sentence before the sentence `The spatial locator for the Block is ...`

If blocks are axially mixed up, relocating each block with the proper axial mesh is performed in ``reestablishBlockOrder``.

Reply:
So the docstring for `reestablishBlockOrder` is misleading, in my opinion. That method actually doesn't do anything with the axial mesh of the assembly. It re-initializes the spatialGrid for the assembly (which confusingly, isn't used to calculate the axial mesh -- see Assembly.getAxialMesh()) and then updates the spatialLocator and name for the Block. 

I will update the impl tag description to reflect this.

Reply:
https://github.com/terrapower/armi/pull/1604/commits/73871c2bee583f47c4276bc0485f235e7c9e05fa